software/webapp/src/main/java/brooklyn/entity/webapp/JavaWebAppSshDriver.java

```diff
@@ -146,15 +149,39 @@ public void deploy(File f, String targetName) {
      * @see JavaWebAppSoftwareProcess#deploy(String, String) for details of how input filenames are handled
      */
     @Override
-    public String deploy(String url, String targetName) {
-        String canonicalTargetName = getFilenameContextMapper().convertDeploymentTargetNameToFilename(targetName);
-        String dest = getDeployDir() + "/" + canonicalTargetName;
-        log.info("{} deploying {} to {}:{}", new Object[]{entity, url, getHostname(), dest});
-        // create a backup
-        getMachine().execCommands("backing up old war", ImmutableList.of(String.format("mv -f %s %s.bak > /dev/null 2>&1", dest, dest))); //back up old file/directory
-        int result = copyResource(url, dest);
-        log.debug("{} deployed {} to {}:{}: result {}", new Object[]{entity, url, getHostname(), dest, result});
-        if (result!=0) log.warn("Problem deploying {} to {}:{} for {}: result {}", new Object[]{url, getHostname(), dest, entity, result}); 
+    public String deploy(final String url, final String targetName) {
+        final String canonicalTargetName = getFilenameContextMapper().convertDeploymentTargetNameToFilename(targetName);
+        final String dest = getDeployDir() + "/" + canonicalTargetName;
+        //write to a .tmp so autodeploy is not triggered during upload
+        final String tmpDest = dest + ".tmp";
```

Comment:
make this `dest+"."+Strings.makeRandomId(8)+".tmp"` ?

that should prevent collisions if the same thing is deployed simultaneously